### PR TITLE
disable deprecated actions

### PR DESCRIPTION
**Pull request by StephenHogsten**
## Description
Set config templates to disable `action_get_macros` and `fn_actions` by default for [#3947](https://github.com/qmk/qmk_firmware/issues/3947)

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [x] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* #3947 

## Checklist

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
   - not sure, it may be worth mentioning the default under 'features that can be disabled'
- [ ] I have updated the documentation accordingly.
- [x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by bcat** [comment]
I think you need to guard these with `#ifdef`s, or else `LINK_TIME_OPTIMIZATION_ENABLE` (which forces these defs on in the makefiles) will break the build. (I just fixed this for Lily58 in #7181.)

File: quantum/template/avr/config.h [hunk 1]
```diff
@@ -190,8 +190,10 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 //#define NO_ACTION_LAYER
 //#define NO_ACTION_TAPPING
 //#define NO_ACTION_ONESHOT
-//#define NO_ACTION_MACRO
-//#define NO_ACTION_FUNCTION
+
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
```
Comment (fauxpark) [inline quantum/template/avr/config.h]:
```suggestion
#ifndef LINK_TIME_OPTIMIZATION_ENABLE
#    define NO_ACTION_MACRO
```

File: quantum/template/avr/config.h [hunk 2]
```diff
@@ -190,8 +190,10 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+#define NO_ACTION_FUNCTION
```
Comment (fauxpark) [inline quantum/template/avr/config.h]:
```suggestion
#    define NO_ACTION_FUNCTION
#endif
```

File: quantum/template/ps2avrgb/config.h [hunk 3]
```diff
@@ -44,6 +44,10 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
 
 #define NO_UART 1
 
+/* disable these deprecated features by default */
+#define NO_ACTION_MACRO
```
Comment (fauxpark) [inline quantum/template/ps2avrgb/config.h]:
```suggestion
#ifndef LINK_TIME_OPTIMIZATION_ENABLE
#    define NO_ACTION_MACRO
```

File: quantum/template/ps2avrgb/config.h [hunk 4]
```diff
@@ -44,6 +44,10 @@ along with this program.  If not, see <http://www.gnu.org/licenses/>.
...
+#define NO_ACTION_FUNCTION
```
Comment (fauxpark) [inline quantum/template/ps2avrgb/config.h]:
```suggestion
#    define NO_ACTION_FUNCTION
#endif
```

**Comment by StephenHogsten** [comment]
I misread the suggestion

**Comment by fauxpark** [comment]
Looks good, thanks!

**Pull request merged**